test/endtoend/test_example.py

```diff
@@ -215,6 +248,8 @@ def test_errors_dict_catches_all_differences(uri_left, uri_right):
                 }
             }
         },
+        'enums': {
+        },
```

Comment:
(TIL) Comment support is new in SQLAlchemy 1.2.0: http://docs.sqlalchemy.org/en/latest/changelog/migration_12.html#change-1546

The change adds a `dialect.supports_comments`.

I think the most straightforward way to go here is to key off that attribute (e.g., `getattr(dialect, 'supports_comments', False)`) and add `'comment': None` to the expected errors when supported.

Perhaps, though, `test_errors_dict_catches_all_differences` should compare `diff` results more selectively? Check that a given key's value differs as expected between left and right, and that the rest match. That would ignore the matching, but unexpected, `"comment"`.

That selectivity should also help make the tests work for a dialect other than mysql (e.g., postgresql), though left_only and right_only would still fail without further work.

Your thoughts?

Reply:
Agree it would be better to be more selective in the assertions rather than comparing the whole result. But if that's too big a change, your first suggestion will do the job well enough for now.

Reply:
For now, I've implemented the simpler change. I didn't want the more-directed tests to hold up this PR.